### PR TITLE
[fnc_analysis] Check whether _sectorData is an array

### PR DESCRIPTION
**Pull request by alef**
Fixes this error message:
```
17:21:44 Error in expression <ectorData];
if("active" in (_sectorData select 1)) then {
[_sector, "data", ["ac>
17:21:44   Error position: <select 1)) then {
[_sector, "data", ["ac>
17:21:44   Error Generic error in expression
17:21:44 File x\alive\addons\fnc_analysis\fnc_gridAnalysisActive.sqf, line 48
```

### CODE REVIEW

**Comment by SpyderBlack723** [comment]
That error message is usually caused by a faulty map index

File: addons/fnc_analysis/fnc_gridAnalysisActive.sqf [hunk 1]
```diff
@@ -45,8 +45,10 @@ _updatedSectors = [];
 	
 	_sectorData = _sector select 2 select 0; //[_sector, "data"] call ALIVE_fnc_sector;
 
-	if("active" in (_sectorData select 1)) then {
-		[_sector, "data", ["active",[]]] call ALIVE_fnc_sector;
+	if("ARRAY" == typeName _sectorData) then {
+		if("active" in (_sectorData select 1)) then {
```
Comment (Glowbal) [inline addons/fnc_analysis/fnc_gridAnalysisActive.sqf]: Both if statements can be merged with lazy eval.

**Comment by alef** [comment]
Yes, I'm testing ALiVE persistence on [VR](https://github.com/alef/15BN/tree/ALiVE-lite/MPMissions/ALiVE.VR)

File: addons/fnc_analysis/fnc_gridAnalysisActive.sqf (hunk 1 above)
Comment (alef) [inline addons/fnc_analysis/fnc_gridAnalysisActive.sqf]: Done.

File: addons/fnc_analysis/fnc_gridAnalysisActive.sqf [hunk 2]
```diff
@@ -45,10 +45,8 @@ _updatedSectors = [];
 	
 	_sectorData = _sector select 2 select 0; //[_sector, "data"] call ALIVE_fnc_sector;
 
-	if("ARRAY" == typeName _sectorData) then {
-		if("active" in (_sectorData select 1)) then {
-			[_sector, "data", ["active",[]]] call ALIVE_fnc_sector;
-		};
+	if("ARRAY" == typeName _sectorData && "active" in (_sectorData select 1)) then {
```
Comment (Glowbal) [inline addons/fnc_analysis/fnc_gridAnalysisActive.sqf]: To use lazy eval in sqf, you will need to wrap a condition with `{}`. 

```sqf
if ("ARRAY" == typeName _sectorData && {"active" in (_sectorData select 1)}) then {
```

**Pull request merged**